blivet/blivet.py

```diff
@@ -460,31 +425,17 @@ def get_free_space(self, disks=None, clear_part_type=None):
         if disks is None:
             disks = self.disks
 
-        if clear_part_type is None:
-            clear_part_type = self.config.clear_part_type
+        if partitions is None:
+            partitions = self.partitions
 
         free = {}
         for disk in disks:
-            should_clear = self.should_clear(disk, clear_part_type=clear_part_type,
-                                             clear_part_disks=[disk.name])
-            if should_clear:
-                free[disk.name] = (disk.size, Size(0))
-                continue
-
             disk_free = Size(0)
             fs_free = Size(0)
             if disk.partitioned:
                 disk_free = disk.format.free
-                for partition in [p for p in self.partitions if p.disk == disk]:
-                    # only check actual filesystems since lvm &c require a bunch of
-                    # operations to translate free filesystem space into free disk
-                    # space
-                    should_clear = self.should_clear(partition,
-                                                     clear_part_type=clear_part_type,
-                                                     clear_part_disks=[disk.name])
-                    if should_clear:
-                        disk_free += partition.size
-                    elif hasattr(partition.format, "free"):
+                for partition in [p for p in partitions if p.disk == disk]:
```

Comment:
This should use parentheses not square brackets (i.e. a generator instead of list).